src/plugins/management/public/components/management_app/management_router.tsx

```diff
@@ -6,56 +6,104 @@
 
 interface ManagementRouterProps {
   history: AppMountParameters['history'];
   dependencies: ManagementAppDependencies;
   setBreadcrumbs: (crumbs?: ChromeBreadcrumb[], appHistory?: ScopedHistory) => void;
-  onAppMounted: (id: string) => void;
   sections: ManagementSection[];
 }
 
-export const ManagementRouter = memo(
-  ({ dependencies, history, setBreadcrumbs, onAppMounted, sections }: ManagementRouterProps) => (
+export const ManagementRouter = ({
+  dependencies,
+  history,
+  setBreadcrumbs,
+  sections,
+}: ManagementRouterProps) => {
+  const [selectedId, setSelectedId] = useState<string>('');
+  const onAppMounted = useCallback((id: string) => {
+    setSelectedId(id);
+    window.scrollTo(0, 0);
+  }, []);
+
+  const solution: KibanaPageTemplateProps['solution'] = {
+    name: i18n.translate('management.nav.label', {
+      defaultMessage: 'Management',
+    }),
+    icon: 'managementApp',
+    id: 'stack-management-nav-header',
+  };
+
+  return (
     <Router history={history}>
-      <EuiPageBody restrictWidth={false} className="mgtPage__body">
-        <Switch>
-          {sections.map((section) =>
-            section
-              .getAppsEnabled()
-              .map((app) => (
-                <Route
-                  path={`${app.basePath}`}
-                  component={() => (
-                    <ManagementAppWrapper
-                      app={app}
-                      setBreadcrumbs={setBreadcrumbs}
-                      onAppMounted={onAppMounted}
-                      history={history}
-                    />
+      <Switch>
+        {sections.map((section) =>
+          section.getAppsEnabled().map((app) => (
+            <Route
+              path={`${app.basePath}`}
+              component={() => (
+                <ManagementAppWrapper
+                  app={app}
+                  setBreadcrumbs={setBreadcrumbs}
+                  onAppMounted={onAppMounted}
+                  history={history}
+                  managementPageLayout={({ children, ...rest }) => (
+                    <KibanaPageTemplate
+                      {...rest}
+                      solution={solution}
+                      pageSideBar={
+                        <ManagementSidebarNav
+                          navId={solution.id}
+                          selectedId={selectedId}
+                          sections={sections}
+                          history={history}
+                        />
+                      }
+                    >
+                      {children}
+                    </KibanaPageTemplate>
                   )}
                 />
-              ))
+              )}
+            />
+          ))
+        )}
+        <Route
+          path={'/'}
+          component={() => (
+            <ManagementLandingPage
+              version={dependencies.kibanaVersion}
+              setBreadcrumbs={setBreadcrumbs}
+              managementPageLayout={({ children, ...rest }) => (
```

Comment:
This is the same way it's handled above with `<ManagementAppWrapper />` and my guess is that it's so that each page doesn't have to remember to accept the entire `solutionNav` configuration.

_I just update this portion a bit to resolve conflicts from a previous PR/_

Reply:
I'm with @cjcenizal on this, including the 'I might be missing something' part. @cchaos seems uncertain so perhaps @myasonik knows.